src/JSInterop/Microsoft.JSInterop/src/JSRuntime.cs

```diff
@@ -172,9 +186,11 @@ internal void EndInvokeJS(long taskId, bool succeeded, ref Utf8JsonReader jsonRe
                 if (succeeded)
                 {
                     var resultType = TaskGenericsUtil.GetTaskCompletionSourceResultType(tcs);
-
-                    var result = JsonSerializer.Deserialize(ref jsonReader, resultType, JsonSerializerOptions);
-                    TaskGenericsUtil.SetTaskCompletionSourceResult(tcs, result);
+                    if (resultType != typeof(VoidReturn))
+                    {
+                        var result = JsonSerializer.Deserialize(ref jsonReader, resultType, JsonSerializerOptions);
+                        TaskGenericsUtil.SetTaskCompletionSourceResult(tcs, result);
+                    }
```

Comment:
Curious, is there a downside to leaving `tcs` hanging here if `resultType` is `VoidReturn`?

Reply:
Ah, yes we definitely do not want to leave `tcs` hanging. This would prevent the caller from knowing when their call had completed.

Seems like we should be setting the result to `null` since that's probably the best representation of the idea of "no value".

Reply:
@watfordgnf - Nice catch! @SteveSandersonMS - Fixed.